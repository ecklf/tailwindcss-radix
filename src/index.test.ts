import path from "path";
import postcss from "postcss";
import tailwind from "tailwindcss";
import { expect, it } from "vitest";
import radix from "./index";

let html = String.raw;

function run(input: string, config: any, plugin = tailwind) {
  let { currentTestName } = expect.getState();

  return postcss(plugin(config)).process(input, {
    from: `${path.resolve(__filename)}?test=${currentTestName}`,
  });
}

it("should generate boolean data attribute variants", async () => {
  let config = {
    content: [
      {
        raw: html`
          <div class="radix-disabled:opacity-50" />
          <div class="group">
            <div class="group-radix-disabled:cursor-not-allowed" />
          </div>
          <div class="group/named">
            <div class="group-radix-disabled/named:cursor-not-allowed" />
          </div>
          <div class="peer" />
          <div class="peer-radix-disabled:cursor-not-allowed" />
          <div class="peer/named" />
          <div class="peer-radix-disabled/named:cursor-not-allowed" />
        `,
      },
    ],
    plugins: [radix],
  };

  return run("@tailwind utilities", config).then((result) => {
    expect(result.css).toMatchInlineSnapshot(`
      ".radix-disabled\\:opacity-50[data-disabled] {
          opacity: 0.5
      }
      .group\\/named[data-disabled] .group-radix-disabled\\/named\\:cursor-not-allowed {
          cursor: not-allowed
      }
      .group[data-disabled] .group-radix-disabled\\:cursor-not-allowed {
          cursor: not-allowed
      }
      .peer\\/named[data-disabled] ~ .peer-radix-disabled\\/named\\:cursor-not-allowed {
          cursor: not-allowed
      }
      .peer[data-disabled] ~ .peer-radix-disabled\\:cursor-not-allowed {
          cursor: not-allowed
      }"
    `);
  });
});

it("should generate value data attribute variants", async () => {
  let config = {
    content: [
      {
        raw: html`
          <div class="radix-state-open:opacity-50" />
          <div class="group">
            <div class="group-radix-state-open:cursor-not-allowed" />
          </div>
          <div class="group/named">
            <div class="group-radix-state-open/named:cursor-not-allowed" />
          </div>
          <div class="peer" />
          <div class="peer-radix-state-open:cursor-not-allowed" />
          <div class="peer/named" />
          <div class="peer-radix-state-open/named:cursor-not-allowed" />
        `,
      },
    ],
    plugins: [radix],
  };

  return run("@tailwind utilities", config).then((result) => {
    expect(result.css).toMatchSnapshot();
  });
});

it("should generate [width|height] utilities", async () => {
  let config = {
    content: [
      {
        raw: html`
          <div class="w-radix-navigation-menu-viewport" />
          <div class="h-radix-navigation-menu-viewport" />
          <div class="w-radix-accordion-content" />
          <div class="h-radix-accordion-content" />
          <div class="w-radix-collapsible-content" />
          <div class="h-radix-collapsible-content" />
        `,
      },
    ],
    plugins: [radix],
  };

  return run("@tailwind utilities", config).then((result) => {
    expect(result.css).toMatchSnapshot();
  });
});

it("should generate [width|height] `content-available` and `trigger` utilities", async () => {
  let config = {
    content: [
      {
        raw: html`
          <div class="w-radix-context-menu" />
          <div class="max-w-radix-context-menu-content-available" />
          <div class="h-radix-context-menu-content-available" />
          <div class="max-h-radix-context-menu-content-available" />
          <div class="w-radix-context-menu-trigger" />
          <div class="h-radix-context-menu-trigger" />

          <div class="w-radix-dropdown-menu" />
          <div class="max-w-radix-dropdown-menu-content-available" />
          <div class="h-radix-dropdown-menu-content-available" />
          <div class="max-h-radix-dropdown-menu-content-available" />
          <div class="w-radix-dropdown-menu-trigger" />
          <div class="h-radix-dropdown-menu-trigger" />

          <div class="w-radix-hover-card" />
          <div class="max-w-radix-hover-card-content-available" />
          <div class="h-radix-hover-card-content-available" />
          <div class="max-h-radix-hover-card-content-available" />
          <div class="w-radix-hover-card-trigger" />
          <div class="h-radix-hover-card-trigger" />

          <div class="w-radix-menubar" />
          <div class="max-w-radix-menubar-content-available" />
          <div class="h-radix-menubar-content-available" />
          <div class="max-h-radix-menubar-content-available" />
          <div class="w-radix-menubar-trigger" />
          <div class="h-radix-menubar-trigger" />

          <div class="w-radix-popover" />
          <div class="max-w-radix-popover-content-available" />
          <div class="h-radix-popover-content-available" />
          <div class="max-h-radix-popover-content-available" />
          <div class="w-radix-popover-trigger" />
          <div class="h-radix-popover-trigger" />

          <div class="w-radix-select" />
          <div class="max-w-radix-select-content-available" />
          <div class="h-radix-select-content-available" />
          <div class="max-h-radix-select-content-available" />
          <div class="w-radix-select-trigger" />
          <div class="h-radix-select-trigger" />

          <div class="w-radix-tooltip" />
          <div class="max-w-radix-tooltip-content-available" />
          <div class="h-radix-tooltip-content-available" />
          <div class="max-h-radix-tooltip-content-available" />
          <div class="w-radix-tooltip-trigger" />
          <div class="h-radix-tooltip-trigger" />
        `,
      },
    ],
    plugins: [radix],
  };

  return run("@tailwind utilities", config).then((result) => {
    expect(result.css).toMatchSnapshot();
  });
});

it("should generate `content-transform-origin` utilities", async () => {
  let config = {
    content: [
      {
        raw: html`
          <div class="origin-radix-context-menu" />
          <div class="origin-radix-dropdown-menu" />
          <div class="origin-radix-hover-card" />
          <div class="origin-radix-menubar" />
          <div class="origin-radix-popover" />
          <div class="origin-radix-select" />
          <div class="origin-radix-tooltip" />
        `,
      },
    ],
    plugins: [radix],
  };

  return run("@tailwind utilities", config).then((result) => {
    expect(result.css).toMatchSnapshot();
  });
});

it("should generate tooltip transform utilities", async () => {
  let config = {
    content: [
      {
        raw: html`
          <div class="translate-x-radix-toast-swipe-end-x" />
          <div class="translate-y-radix-toast-swipe-end-y" />
          <div class="translate-x-radix-toast-swipe-move-x" />
          <div class="translate-y-radix-toast-swipe-move-y" />
        `,
      },
    ],
    plugins: [radix],
  };

  return run("@tailwind utilities", config).then((result) => {
    expect(result.css).toMatchSnapshot();
  });
});
