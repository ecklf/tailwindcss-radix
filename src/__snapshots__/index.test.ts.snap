// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should generate [width|height] \`content-available\` utilities 1`] = `
".w-radix-context-menu-content-available {
    width: var(--radix-context-menu-content-available-width)
}
.max-w-radix-context-menu-content-available {
    max-width: var(--radix-context-menu-content-available-width)
}
.h-radix-context-menu-content-available {
    height: var(--radix-context-menu-content-available-height)
}
.max-h-radix-context-menu-content-available {
    max-height: var(--radix-context-menu-content-available-height)
}
.w-radix-dropdown-menu-content-available {
    width: var(--radix-dropdown-menu-content-available-width)
}
.max-w-radix-dropdown-menu-content-available {
    max-width: var(--radix-dropdown-menu-content-available-width)
}
.h-radix-dropdown-menu-content-available {
    height: var(--radix-dropdown-menu-content-available-height)
}
.max-h-radix-dropdown-menu-content-available {
    max-height: var(--radix-dropdown-menu-content-available-height)
}
.w-radix-hover-card-content-available {
    width: var(--radix-hover-card-content-available-width)
}
.max-w-radix-hover-card-content-available {
    max-width: var(--radix-hover-card-content-available-width)
}
.h-radix-hover-card-content-available {
    height: var(--radix-hover-card-content-available-height)
}
.max-h-radix-hover-card-content-available {
    max-height: var(--radix-hover-card-content-available-height)
}
.w-radix-menubar-content-available {
    width: var(--radix-menubar-content-available-width)
}
.max-w-radix-menubar-content-available {
    max-width: var(--radix-menubar-content-available-width)
}
.h-radix-menubar-content-available {
    height: var(--radix-menubar-content-available-height)
}
.max-h-radix-menubar-content-available {
    max-height: var(--radix-menubar-content-available-height)
}
.w-radix-popover-content-available {
    width: var(--radix-popover-content-available-width)
}
.max-w-radix-popover-content-available {
    max-width: var(--radix-popover-content-available-width)
}
.h-radix-popover-content-available {
    height: var(--radix-popover-content-available-height)
}
.max-h-radix-popover-content-available {
    max-height: var(--radix-popover-content-available-height)
}
.w-radix-select-content-available {
    width: var(--radix-select-content-available-width)
}
.max-w-radix-select-content-available {
    max-width: var(--radix-select-content-available-width)
}
.h-radix-select-content-available {
    height: var(--radix-select-content-available-height)
}
.max-h-radix-select-content-available {
    max-height: var(--radix-select-content-available-height)
}
.w-radix-tooltip-content-available {
    width: var(--radix-tooltip-content-available-width)
}
.max-w-radix-tooltip-content-available {
    max-width: var(--radix-tooltip-content-available-width)
}
.h-radix-tooltip-content-available {
    height: var(--radix-tooltip-content-available-height)
}
.max-h-radix-tooltip-content-available {
    max-height: var(--radix-tooltip-content-available-height)
}"
`;

exports[`should generate [width|height] \`trigger\` utilities 1`] = `
".w-radix-context-menu-trigger {
    width: var(--radix-context-menu-trigger-width)
}
.h-radix-context-menu-trigger {
    height: var(--radix-context-menu-trigger-height)
}
.w-radix-dropdown-menu-trigger {
    width: var(--radix-dropdown-menu-trigger-width)
}
.h-radix-dropdown-menu-trigger {
    height: var(--radix-dropdown-menu-trigger-height)
}
.w-radix-hover-card-trigger {
    width: var(--radix-hover-card-trigger-width)
}
.h-radix-hover-card-trigger {
    height: var(--radix-hover-card-trigger-height)
}
.w-radix-menubar-trigger {
    width: var(--radix-menubar-trigger-width)
}
.h-radix-menubar-trigger {
    height: var(--radix-menubar-trigger-height)
}
.w-radix-popover-trigger {
    width: var(--radix-popover-trigger-width)
}
.h-radix-popover-trigger {
    height: var(--radix-popover-trigger-height)
}
.w-radix-select-trigger {
    width: var(--radix-select-trigger-width)
}
.h-radix-select-trigger {
    height: var(--radix-select-trigger-height)
}
.w-radix-tooltip-trigger {
    width: var(--radix-tooltip-trigger-width)
}
.h-radix-tooltip-trigger {
    height: var(--radix-tooltip-trigger-height)
}"
`;

exports[`should generate [width|height] utilities 1`] = `
".w-radix-accordion-content {
    width: var(--radix-accordion-content-width)
}
.h-radix-accordion-content {
    height: var(--radix-accordion-content-height)
}
.w-radix-collapsible-content {
    width: var(--radix-collapsible-content-width)
}
.h-radix-collapsible-content {
    height: var(--radix-collapsible-content-height)
}
.w-radix-navigation-menu-viewport {
    width: var(--radix-navigation-menu-viewport-width)
}
.h-radix-navigation-menu-viewport {
    height: var(--radix-navigation-menu-viewport-height)
}"
`;

exports[`should generate \`content-transform-origin\` utilities 1`] = `
".origin-radix-context-menu {
    transform-origin: var(--radix-context-menu-content-transform-origin)
}
.origin-radix-dropdown-menu {
    transform-origin: var(--radix-dropdown-menu-content-transform-origin)
}
.origin-radix-hover-card {
    transform-origin: var(--radix-hover-card-content-transform-origin)
}
.origin-radix-menubar {
    transform-origin: var(--radix-menubar-content-transform-origin)
}
.origin-radix-popover {
    transform-origin: var(--radix-popover-content-transform-origin)
}
.origin-radix-select {
    transform-origin: var(--radix-select-content-transform-origin)
}
.origin-radix-tooltip {
    transform-origin: var(--radix-tooltip-content-transform-origin)
}"
`;

exports[`should generate tooltip transform utilities 1`] = `
".translate-x-radix-toast-swipe-end-x {
    transform: translateX(var(--radix-toast-swipe-end-x))
}
.translate-y-radix-toast-swipe-end-y {
    transform: translateY(var(--radix-toast-swipe-end-y))
}
.translate-x-radix-toast-swipe-move-x {
    transform: translateX(var(--radix-toast-swipe-move-x))
}
.translate-y-radix-toast-swipe-move-y {
    transform: translateY(var(--radix-toast-swipe-move-y))
}"
`;

exports[`should generate value data attribute variants 1`] = `
".radix-state-open\\:opacity-50[data-state="open"] {
    opacity: 0.5
}
.group\\/named[data-state="open"] .group-radix-state-open\\/named\\:cursor-not-allowed {
    cursor: not-allowed
}
.group[data-state="open"] .group-radix-state-open\\:cursor-not-allowed {
    cursor: not-allowed
}
.peer\\/named[data-state="open"] ~ .peer-radix-state-open\\/named\\:cursor-not-allowed {
    cursor: not-allowed
}
.peer[data-state="open"] ~ .peer-radix-state-open\\:cursor-not-allowed {
    cursor: not-allowed
}"
`;
